// Copyright 2020-2022 CYBERCRYPT

syntax = "proto3";

package d1.version;
option go_package = "github.com/cybercryptio/d1-service-generic/v2/protobuf/version";
option csharp_namespace = "CyberCrypt.D1.Client.Protobuf.Version";
option java_package = "io.cybercrypt.d1.protobuf.version";
option java_outer_classname = "VersionProto";

// Service for getting version information.
service Version {
  // Returns the versions of the currently running service.
  rpc Version (VersionRequest) returns (VersionResponse) {}
}

// Represents a request to get version information.
message VersionRequest {
}

// Represents a response to get version information.
message VersionResponse {
  // The Git commit of the running service.
  string commit = 1;
  // The version of the running service.
  string tag = 2;
}
